services/src/main/groovy/org/jfrog/artifactory/client/ArtifactoryClientBuilder.java

```diff
@@ -95,6 +98,22 @@ public ArtifactoryClientBuilder setAccessToken(String accessToken) {
         return this;
     }
 
+    /**
+     * Add an Http request interceptor to the underlying Http client builder used by the artifactory client
+     * <br>
+     * For further details see
+     * {@link org.apache.http.impl.client.HttpClientBuilder#addInterceptorLast(org.apache.http.HttpRequestInterceptor)}
+     * @param httpRequestInterceptor request interceptor that allows manipulating and examining of outgoing requests
+     * @return ArtifactoryClientBuilder
+     */
+    public ArtifactoryClientBuilder addInterceptorLast(HttpRequestInterceptor httpRequestInterceptor) {
+        if (this.requestInterceptorList == null) {
+            this.requestInterceptorList = new ArrayList<>();
```

Comment:
I suggest to add an initialization in the declaration, instead of this `if` and the `if` in line 148:
private List<HttpRequestInterceptor> requestInterceptorLis = new ArrayList<>();

Reply:
ok, I will fix accordingly